core/psh/ping/ping.c

```diff
@@ -167,120 +167,170 @@ static int psh_ping_recv(int fd, uint8_t *data, int len, char *infostr, int info
 		return -1;
 	}
 
-	iphdr = (struct iphdr *) data;
-	icmphdr = (struct icmphdr *) (data + sizeof(struct iphdr));
+	icmphdr = (struct icmphdr *)(data + sizeof(struct iphdr));
```

Comment:
probably ok but looks :fish: 

side note: should we validate if response size matches the request size? (is it defined in ICMP protocol?)

Reply:
Yes, as You said, it can be made nicer by defining a structure "struct packet".
When I was experimenting with different payload sizes it occured, that very often, when my payload size was larger than ~64 bytes I was receiving smaller response. It might be specific to network or peer. Or maybe lwip limits the size of the payload. I'm not sure.

Reply:
ok, let's leave it as-is for now.